mlrun/projects/project.py

```diff
@@ -3446,6 +3448,128 @@ def remove_custom_packager(self, packager: str):
         """
         self.spec.remove_custom_packager(packager=packager)
 
+    def create_api_gateway(
+        self,
+        name: str,
+        functions: Union[
+            list[
+                Union[
+                    RemoteRuntime,
+                    ServingRuntime,
+                ]
+            ],
+            Union[RemoteRuntime, ServingRuntime],
+        ],
+        path: Optional[str] = "/",
+        description: Optional[str] = "",
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        canary: Optional[list[int]] = None,
+    ) -> Union[mlrun.runtimes.api_gateway.APIGateway, None]:
+        """
+        Creates a Nuclio API Gateway. Nuclio docs here: https://docs.nuclio.io/en/latest/reference/api-gateway/http.html
+
+        :param name (str): API Gateway name.
+        :param functions (Union[list[Union[RemoteRuntime, ServingRuntime]], Union[RemoteRuntime, ServingRuntime]]):
+            List of Nuclio functions or Nuclio function.
+        :param host (str): API Gateway host name (optional). If not specified, nuclio will automatically set it
+            to `<api-gateway-name>-<project>.<system-domain>`
```

Comment:
@TomerShor I fully deleted `host` param on MLRun client side (will clean up docs as well), it is being generated on server side now. It was done like this because user doesn't have `nuclio_dashboard` info on client side.